File: models/Request.js
const mongoose = require('mongoose');

// ==========================================
// [THÊM MỚI]: Cấu trúc của một Bình luận
// ==========================================
const commentSchema = new mongoose.Schema({
  user: { 
    type: String, 
    ref: 'User', 
    required: true 
  }, // Lưu UID của người bình luận
  content: { 
    type: String, 
    required: [true, 'Nội dung bình luận không được để trống'] 
  },
  createdAt: { 
    type: Date, 
    default: Date.now 
  }
});

const requestSchema = new mongoose.Schema(
  {
    author: {
      type: String, // Trùng khớp với UID của Firebase
      ref: 'User',
      required: true,
    },
    title: {
      type: String,
      required: [true, 'Vui lòng nhập tiêu đề yêu cầu'],
      trim: true,
    },
    description: {
      type: String,
      required: [true, 'Vui lòng nhập mô tả chi tiết'],
      trim: true,
    },
    status: {
      type: String,
      enum: ['open', 'resolved', 'closed'], 
      default: 'open',
    },
    // Danh sách người dùng (UID) đã upvote
    upvotes: [{
      type: String,
      ref: 'User'
    }],
    // Link tài liệu khi có người tìm thấy
    resolvedLink: {
      type: String,
      default: null
    },
    // Người đã cung cấp tài liệu đó
    resolvedBy: {
      type: String,
      ref: 'User',
      default: null
    },
    // ==========================================
    // [THÊM MỚI]: Mảng chứa danh sách các bình luận thảo luận
    // ==========================================
    comments: [commentSchema]
  },
  {
    timestamps: true, // Tự động tạo createdAt và updatedAt
  }
);

module.exports = mongoose.model('Request', requestSchema);

File: app/src/main/java/com/example/thuctaptotnghiep/data/model/Request.kt
package com.example.thuctaptotnghiep.data.model

import com.google.gson.annotations.SerializedName

data class Request(
    @SerializedName("_id") val id: String, // Chuẩn hóa từ _id sang id
    @SerializedName("author") val author: User?,
    @SerializedName("title") val title: String,
    @SerializedName("description") val description: String,
    @SerializedName("status") val status: String, // "open", "resolved", "closed"
    @SerializedName("createdAt") val createdAt: String,

    // ============================================================
    // [THÊM MỚI] - CÁC TRƯỜNG PHỤC VỤ CHO LUỒNG DIỄN ĐÀN THẢO LUẬN
    // ============================================================
    @SerializedName("upvotes") val upvotes: List<String> = emptyList(), // Danh sách UID đã upvote
    @SerializedName("resolvedLink") val resolvedLink: String? = null,
    @SerializedName("resolvedBy") val resolvedBy: User? = null,
    @SerializedName("comments") val comments: List<Comment> = emptyList() // Danh sách các bình luận
)

// Cấu trúc dữ liệu của một Bình luận trong bài viết công đồng
data class Comment(
    @SerializedName("_id") val id: String,
    @SerializedName("user") val user: User?, // Thông tin người bình luận (Đã được BE tự động populate)
    @SerializedName("content") val content: String,
    @SerializedName("createdAt") val createdAt: String
)